scripts/migrate-tags.md

```diff
@@ -0,0 +1,158 @@
+# `migrate-tags`
+
+## A. Preparations
+
+- The migration target package must be inside of the `merged-packages/` directory with its git history fully migrated.
+- The script must be run from the root directory of the core repo.
+- If the script isn't executable, run `chmod +x ./scripts/migrate-tags.sh`.
+- By default, this script will run in "dry mode", printing out all pairs of release commit hashes and prefixed tag names, but not modifying the local or remote repo in any way. To override this and actually create/push tags, run the script with a `--non-dry-run` flag appended at the end.
+
+## B. Options
+
+- `<package-name>` (required).
+  - Only supply the package directory name. Exclude the `@metamask/` namespace.
+- `-r`, `--remote` (optional): the git remote repo where the tags will be pushed.
+  - Default if omitted: "origin".
+- `-p`, `--regex-pattern` (optional): regex pattern for grepping release commits from the package's migrated git history.
+  - Default if omitted: regex for commit messages that start with a semver string.
+- `-t`, `--tag-prefix-before-package-rename` (optional)
+  - Default if omitted: `<package-name>` supplied in the first argument.
+- `-v`, `--version-before-package-rename` (optional)
+  - Default if omitted: `0.0.0`.
```

Comment:
This default works because of https://github.com/MetaMask/core/pull/1802/files#diff-97de9f884bc1bbb04c3ae63fba81d4f66e8d39146bcda0d5ec2887686285238cR56-R59. If `<version-before-package-rename>` is `0.0.0`, tags are named with the `@metamask/<package-name>@<tag-version>` scheme, and the package rename-handling logic is never invoked. So omitting `-v` is essentially declaring that there is no package rename to worry about.